src/Adapter/CrudTrait.php

```diff
@@ -68,22 +74,44 @@ public function update($table, array $data, $where = '', array $bind = [])
      * Delete from table.
      *
      * @param string $table
-     * @param string $where
-     * @param array $bind
+     * @param array|string $where Deprecated usage of parameter as string
+     * @param array $bind Deprecated in favour of $where as array
      * @return int Number of affected rows
      */
-    public function delete($table, $where = '', array $bind = [])
+    public function delete($table, $where = [], array $bind = [])
     {
         $table = $this->quote()->identifier($table);
-        $sql   = "DELETE FROM $table"
-            . (($where) ? " WHERE $where" : '');
+        $sql   = "DELETE FROM $table";
+        if (is_array($where)) {
+            $where = $this->createWhereExpression($where);
+        }
+        $sql .= (($where) ? " WHERE {$where}" : '');
 
         $stmt = $this->query($sql, $bind);
 
         return $stmt->rowCount();
     }
 
     /**
+     * Create WHERE expression from given criteria
+     *
+     * @param array $where
+     * @return string
+     */
+    private function createWhereExpression(array $where = [])
+    {
+        $criteria = [];
+        foreach ($where as $str => $value) {
+            if (is_int($str)) {
```

Comment:
Shouldn't the quoter take care of this scenario? QuoteHandler line 51.

Reply:
I don't think so. This is testing if the array element is `0 => 'col2 IS NULL'` or `'col2 = ?' => 'abc'`.
Like [ZF1](https://github.com/zendframework/zf1/blob/master/library/Zend/Db/Adapter/Abstract.php#L686)

Reply:
Oh I see, I misread the code. I think it would be better if the variable `$str` is named `$index`.